<!-- 
The MIT License (MIT)
Copyright (c) 2016 GRVC University of Seville

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE. 
-->
<launch>

    <arg name="id" default="1"/>
    <arg name="ual_server" default="on"/>
    <arg name="ual_pub_rate" default="30.0"/>

    <arg name="pose_frame_id" default=""/><!-- To publish ual/pose in other frame than local, e.g. map -->
    <arg name="dji_offboard_rate" default="30.0"/>

    <arg name="position_th" default="0.33"/>
    <arg name="orientation_th" default="0.6"/>
    <arg name="vel_factor" default="0.7"/>
    <arg name="laser_altimeter" default="false"/>
    <arg name="self_arming" default="false"/>
    <arg name="xy_vel_max" default="2.3"/>
    <arg name="z_vel_max_up" default="2.0"/>
    <arg name="z_vel_max_dn" default="2.0"/>
    <arg name="yawrate_max" default="0.8"/>

    <!-- Launch DJI-ROS node -->
    <node pkg="dji_sdk" type="dji_sdk_node" name="dji_sdk" output="screen">
        <param name="serial_name" type="string" value="/dev/ttyUSB0"/>
        <param name="baud_rate" type="int" value="921600"/>
        <param name="app_id" type="int" value="your app id here"/>
        <param name="app_version" type="int" value="1"/>
        <param name="align_time" type="bool" value="false"/>
        <param name="enc_key" type="string" value="your app key here"/>
        <param name="use_broadcast" type="bool" value="false"/>
    </node>

    <!-- Launch UAL server -->
    <node pkg="ual_backend_dji_ros" type="ual_backend_dji_ros_server" name="ual" output="screen">
        <param name="uav_id" value="$(arg id)"/>
        <param name="ual_server" value="$(arg ual_server)"/>
        <param name="ual_pub_rate" value="$(arg ual_pub_rate)"/>
        <param name="pose_frame_id" value="$(arg pose_frame_id)"/>
        <param name="dji_offboard_rate" value="$(arg dji_offboard_rate)"/>
        <param name="position_th" value="$(arg position_th)"/>
        <param name="orientation_th" value="$(arg orientation_th)"/>
        <param name="vel_factor" value="$(arg vel_factor)"/>
        <param name="laser_altimeter" value="$(arg laser_altimeter)"/>
        <param name="self_arming" value="$(arg self_arming)"/>
        <param name="xy_vel_max" value="$(arg xy_vel_max)"/>
        <param name="z_vel_max_up" value="$(arg z_vel_max_up)"/>
        <param name="z_vel_max_dn" value="$(arg z_vel_max_dn)"/>
        <param name="yawrate_max" value="$(arg yawrate_max)"/>
        <param name="home_pose_parent_frame" value="map"/><!-- default= map -->
        <!-- Define home pose relative to global map frame -->
        <rosparam param="home_pose">[0, 0, 0]</rosparam><!-- [x,y,z] -->
        <!-- Define the global map frame geo position and let the UAL estimate the relative home pose -->
        <!-- <rosparam param="map_origin_geo">[37.558542, -5.931074, 7.89]</rosparam> --><!-- [lat,lon,alt] -->
    </node>

</launch>
